lib/ret.ex

```diff
@@ -16,4 +21,76 @@ defmodule Ret do
       {:error, :forbidden}
     end
   end
+
+  defp delete_avatars_for_account(%Account{} = account) do
+    account_avatars_query = from(avatar in Avatar, where: avatar.account_id == ^account.account_id)
+    account_avatars = Repo.all(account_avatars_query) |> Repo.preload(Avatar.file_columns())
+
+    account_avatar_owned_files =
+      account_avatars
+      |> Enum.flat_map(fn avatar ->
+        Avatar.file_columns()
+        |> Enum.map(fn column -> Map.fetch!(avatar, column) end)
+        |> Enum.filter(fn owned_file -> owned_file != nil end)
+      end)
+
+    Repo.delete_all(account_avatars_query)
+    delete_owned_files(account_avatar_owned_files)
+  end
+
+  defp delete_owned_files(owned_files) when is_list(owned_files) do
+    for owned_file <- owned_files, do: OwnedFile.set_inactive(owned_file)
+    for owned_file <- owned_files, do: Storage.rm_files_for_owned_file(owned_file)
+    for owned_file <- owned_files, do: Repo.delete(owned_file)
```

Comment:
Is there any reason we couldn’t do this in one iteration?

```elixir
for owned_file <- owned_files do
  OwnedFile.set_inactive(owned_file)
  Storage.rm_files_for_owned_file(owned_file)
  Repo.delete(owned_file)
end
```

Reply:
Initially I figured it would be better to maintain an order of operations, where we first ensure that all the files are inactive, and cannot be accessed, then we delete the physical files on disk, before we delete all the DB records. But I suppose there's no harm in doing them one by one and it is probably more performant.